universal-application-tool-0.0.1/test/services/program/ProgramServiceImplTest.java

```diff
@@ -552,4 +552,41 @@ public void setBlockOptionalPredicate_constructsQuestionDefinitions()
         found.blockDefinitions().get(0).programQuestionDefinitions().get(0).getQuestionDefinition();
     assertThat(foundQuestion).isInstanceOf(NameQuestionDefinition.class);
   }
+
+  @Test
+  public void deleteBlock_invalidProgram_throwsProgramNotfoundException() {
+    assertThatThrownBy(() -> ps.deleteBlock(1L, 2L))
+        .isInstanceOf(ProgramNotFoundException.class)
+        .hasMessage("Program not found for ID: 1");
+  }
+
+  @Test
+  public void deleteBlock_constructsQuestionDefinitions() throws ProgramNotFoundException {
+    QuestionDefinition question = qs.create(SIMPLE_QUESTION).get();
+    ProgramDefinition programDefinition =
+        ps.createProgramDefinition("Program With Block", "This program has a block.");
+    Long programId = programDefinition.id();
+    ps.addBlockToProgram(
+        programId,
+        "the block",
+        "the block for the program",
+        ImmutableList.of(ProgramQuestionDefinition.create(question)));
+    ps.addBlockToProgram(
+        programId,
+        "the second block",
+        "the second block for the program",
+        ImmutableList.of(ProgramQuestionDefinition.create(question)));
+
+    ProgramDefinition result = ps.deleteBlock(programId, 1L);
```

Comment:
suggestion (non-blocking): Is there an easy way to get the ID of the first block programmatically? Might be nice for readability, but if it's a pain then this is ok